migrations/utils.py

```diff
@@ -0,0 +1,124 @@
+class Node:
+    def __init__(self, data):
+        self.data = data
+        self.prev_node = None
+        self.next_node = None
+
+    def __repr__(self):
+        return f"{self.data}"
+
+    def __eq__(self, other):
+        if isinstance(other, Node):
+            return self.data == other.data
+        return False
+
+    def __hash__(self):
+        return hash(self.data)
+
+
+class LinkedList:
+    def __init__(self):
+        self.head = None
+
+    def find(self, data) -> Node:
+        """Find the first node containing the specified data."""
+        # Traverse forward
+        current_node = self.head
+        while current_node:
+            if current_node.data == data:
+                return current_node
+            current_node = current_node.next_node
+
+        # If not found, start from the tail node
+        current_node = self.head
+        while current_node:
+            if current_node.data == data:
+                return current_node
+            current_node = current_node.prev_node
+
+        # If not found, return None
+        return None
+
+    def get_following_node(self, current_node_data) -> Node:
+        """Retrieve node following the specified one."""
+        next_node = self.head
+        while next_node and next_node.prev_node:
+            if next_node.prev_node.data == current_node_data:
+                return next_node
+            next_node = next_node.prev_node
+        return None
+
+    def get_previous_node(self, current_node_data) -> Node:
+        """Retrieve node before the specified one."""
+        current_node = self.find(current_node_data)
+        if current_node and current_node.prev_node:
+            return current_node.prev_node.data
+        return None
+
+    def get_sublist(self, first_node_data: str, last_node_data: str = None) -> list:
+        """Return a list consistings of nodes between the specified boundaries.
+        Inclusive of last endpoint."""
+        unapplied_migrations = []
+        if last_node_data is None:
+            last_node = self.head
+        else:
+            last_node = self.find(last_node_data)
+
+        # Iterate over migrations starting from top
+        while last_node and last_node.data != first_node_data:
+            unapplied_migrations.append(last_node.data)
+            last_node = last_node.prev_node
+
+        # Reverse to account for the order
+        unapplied_migrations.reverse()
+
+        return unapplied_migrations
+
+    def append(self, data):
+        """Append node after the head."""
+        new_node = self.find(data)
+        # If do not exist, create the nodes
```

Comment:
wouldn't it be an error, if the node already exists?   i believe the implementation here doesn't handle duplicates.  this should be documented in the class docstring.